scripts/test/system.sh

```diff
@@ -68,6 +68,46 @@ test_subctl_gather
 _subctl diagnose all --validation-timeout 20
 _subctl diagnose firewall inter-cluster --validation-timeout 20 "${KUBECONFIGS_DIR}"/kind-config-cluster1 "${KUBECONFIGS_DIR}"/kind-config-cluster2
 
+# Test subctl diagnose in-cluster
+echo "::group::Validating 'subctl diagnose' in-cluster"
+
+job_name="submariner-diagnose"
+kubectl apply -f - <<EOF
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: ${job_name}
+  namespace: ${subm_ns}
+spec:
+  template:
+    metadata:
+      labels:
+        submariner.io/transient: "true"
+    spec:
+      containers:
+      - name: ${job_name}
+        image: localhost:5000/subctl:local
+        command: ["subctl",  "diagnose", "all", "--in-cluster", "--validation-timeout", "20"]
+      restartPolicy: Never
+      serviceAccount: submariner-diagnose
+      serviceAccountName: submariner-diagnose
+  backoffLimit: 0
+EOF
+
+echo "Wait for job to be ready"
+kubectl wait --for=condition=Ready "job/${job_name}"
+
+echo "Print diagnose logs"
+kubectl logs "job/${job_name}" -n "${subm_ns}" -f
+if kubectl get jobs "${job_name}" -n "${subm_ns}" -o jsonpath='{.status.conditions[?(@.type=="Failed")].status}' | grep "True" > /dev/null
+then
+  kubectl delete job "${job_name}" -n "${subm_ns}"
+  exit 1
```

Comment:
```suggestion
    kubectl delete job "${job_name}" -n "${subm_ns}"
    exit 1
```

For consistency with indentation in the file